pkg/auth/dependency/webapp/authenticate_provider_impl.go

```diff
@@ -40,12 +51,18 @@ func (p *AuthenticateProviderImpl) ServeHTTP(w http.ResponseWriter, r *http.Requ
 	var err error
 	step := r.Form.Get("x_step")
 	switch step {
-	case "submit_password":
-		writeResponse, err = p.SubmitPassword(w, r)
 	case "submit_login_id":
 		writeResponse, err = p.SubmitLoginID(w, r)
+	case "submit_password":
+		writeResponse, err = p.SubmitPassword(w, r)
 	case "choose_idp":
 		writeResponse, err = p.ChooseIdentityProvider(w, r)
+	case "sign_up":
```

Comment:
It seems the step names are getting complex. Maybe add a string enum and add structure to the step names? e.g. `signin:submit_login_id`, `signup:submit_password`.